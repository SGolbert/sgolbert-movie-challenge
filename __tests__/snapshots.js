import React from "react";
import { render, cleanup } from "@testing-library/react";
import Home from "../pages/index";
import CategoryListPage from "../pages/categories/[category_id]/[genre]";
import MoviePage from "../pages/movies/[movie_id]";
import SearchPage from "../pages/search/[query]";

afterEach(cleanup);

it("renders homepage unchanged", () => {
  const categories = [
    { name: "Action", id: "1" },
    { name: "Crime", id: "2" },
  ];

  const { asFragment } = render(<Home categories={categories} />);
  expect(asFragment()).toMatchSnapshot();
});

const movieList = [
  {
    title: "Back to the future",
    id: "1",
    image: "/1234",
  },
  {
    title: "Star Wars",
    id: "2",
    image: "/1234",
  },
];

it("renders category page unchanged", () => {
  const { asFragment } = render(<CategoryListPage movies={movieList} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders search page unchanged", () => {
  const { asFragment } = render(<SearchPage movies={movieList} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders movie page unchanged", () => {
  const movieDetails = {
    title: "Back to the future",
    original_language: "en",
    release_date: "1980-01-01",
    overview: "Great movie!",
    vote_count: 10000,
    vote_average: 10,
    poster_path: "/1234",
  };
  const { asFragment } = render(<MoviePage movie={movieDetails} />);
  expect(asFragment()).toMatchSnapshot();
});
