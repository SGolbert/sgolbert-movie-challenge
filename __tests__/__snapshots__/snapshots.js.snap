// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders category page unchanged 1`] = `
<DocumentFragment>
  <header
    class="jsx-1356211436 header"
  >
    <a
      class="jsx-1356211436"
      href="/"
    >
      Home
    </a>
    <a
      class="jsx-1356211436"
      href="/about"
    >
      About
    </a>
    <form
      class="jsx-1356211436"
    >
      <input
        class="jsx-1356211436 movieQuery"
        placeholder="Search for a movie..."
        type="input"
      />
    </form>
  </header>
  <main
    class="jsx-3550297800"
  >
    <h1
      class="jsx-3550297800"
    >
       movies
    </h1>
    <div
      class="jsx-3550297800 movieContainer"
    >
      <a
        class="jsx-3550297800 movieItem"
        href="/movies/1"
      >
        <img
          class="jsx-3550297800"
          height="513px"
          src="https://image.tmdb.org/t/p/w342/1234"
          width="342px"
        />
        <p
          class="jsx-3550297800 movieTitle"
        >
          Back to the future
        </p>
      </a>
      <a
        class="jsx-3550297800 movieItem"
        href="/movies/2"
      >
        <img
          class="jsx-3550297800"
          height="513px"
          src="https://image.tmdb.org/t/p/w342/1234"
          width="342px"
        />
        <p
          class="jsx-3550297800 movieTitle"
        >
          Star Wars
        </p>
      </a>
    </div>
    <div
      class="jsx-3550297800 pagination"
    >
      <ul
        class="pagination"
      >
        <li
          class="previous disabled"
        >
          <a
            aria-disabled="true"
            role="button"
            tabindex="0"
          >
            previous
          </a>
        </li>
        <li
          class="active"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            role="button"
            tabindex="0"
          >
            1
          </a>
        </li>
        <li
          class="next disabled"
        >
          <a
            aria-disabled="true"
            role="button"
            tabindex="0"
          >
            next
          </a>
        </li>
      </ul>
    </div>
    <button
      class="goback"
    >
      Go back
    </button>
  </main>
  <footer
    class="jsx-627519052 footer"
  >
    Copyright 2020 - Sebastian Golbert
  </footer>
</DocumentFragment>
`;

exports[`renders homepage unchanged 1`] = `
<DocumentFragment>
  <header
    class="jsx-1356211436 header"
  >
    <a
      class="jsx-1356211436"
      href="/"
    >
      Home
    </a>
    <a
      class="jsx-1356211436"
      href="/about"
    >
      About
    </a>
    <form
      class="jsx-1356211436"
    >
      <input
        class="jsx-1356211436 movieQuery"
        placeholder="Search for a movie..."
        type="input"
      />
    </form>
  </header>
  <main
    class="jsx-1091064000"
  >
    <h1
      class="jsx-1091064000 title"
    >
      These are the most popular movies at The Movie Database
    </h1>
    <div
      class="jsx-1091064000 categoryContainer"
    >
      <a
        class="jsx-1091064000 categoryItem"
        href="/categories/1/Action"
      >
        Action
      </a>
      <a
        class="jsx-1091064000 categoryItem"
        href="/categories/2/Crime"
      >
        Crime
      </a>
    </div>
    <div
      class="jsx-1091064000 paginate"
    >
      <ul
        class="pagination"
      >
        <li
          class="previous disabled"
        >
          <a
            aria-disabled="true"
            role="button"
            tabindex="0"
          >
            previous
          </a>
        </li>
        <li
          class="active"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            role="button"
            tabindex="0"
          >
            1
          </a>
        </li>
        <li
          class="next disabled"
        >
          <a
            aria-disabled="true"
            role="button"
            tabindex="0"
          >
            next
          </a>
        </li>
      </ul>
    </div>
  </main>
  <footer
    class="jsx-627519052 footer"
  >
    Copyright 2020 - Sebastian Golbert
  </footer>
</DocumentFragment>
`;

exports[`renders movie page unchanged 1`] = `
<DocumentFragment>
  <header
    class="jsx-1356211436 header"
  >
    <a
      class="jsx-1356211436"
      href="/"
    >
      Home
    </a>
    <a
      class="jsx-1356211436"
      href="/about"
    >
      About
    </a>
    <form
      class="jsx-1356211436"
    >
      <input
        class="jsx-1356211436 movieQuery"
        placeholder="Search for a movie..."
        type="input"
      />
    </form>
  </header>
  <main
    class="jsx-3477680548"
  >
    <h1
      class="jsx-3477680548"
    >
      Back to the future
    </h1>
    <div
      class="jsx-3477680548 movieDataContainer"
    >
      <img
        class="jsx-3477680548"
        src="https://image.tmdb.org/t/p/w500/1234"
      />
      <p
        class="jsx-3477680548"
      >
        <span
          class="jsx-3477680548 detail"
        >
          Language:
        </span>
         en
      </p>
      <p
        class="jsx-3477680548"
      >
        <span
          class="jsx-3477680548 detail"
        >
          Release date:
        </span>
         1980-01-01
      </p>
      <p
        class="jsx-3477680548"
      >
        <span
          class="jsx-3477680548 detail"
        >
          Overview:
        </span>
         Great movie!
      </p>
      <p
        class="jsx-3477680548"
      >
        <span
          class="jsx-3477680548 detail"
        >
          Number of votes:
        </span>
         10000
      </p>
      <p
        class="jsx-3477680548"
      >
        <span
          class="jsx-3477680548 detail"
        >
          Average rating:
        </span>
         10
      </p>
    </div>
    <button
      class="goback"
    >
      Go back
    </button>
  </main>
  <footer
    class="jsx-627519052 footer"
  >
    Copyright 2020 - Sebastian Golbert
  </footer>
</DocumentFragment>
`;

exports[`renders search page unchanged 1`] = `
<DocumentFragment>
  <header
    class="jsx-1356211436 header"
  >
    <a
      class="jsx-1356211436"
      href="/"
    >
      Home
    </a>
    <a
      class="jsx-1356211436"
      href="/about"
    >
      About
    </a>
    <form
      class="jsx-1356211436"
    >
      <input
        class="jsx-1356211436 movieQuery"
        placeholder="Search for a movie..."
        type="input"
      />
    </form>
  </header>
  <main
    class="jsx-3550297800"
  >
    <h1
      class="jsx-3550297800"
    >
      Search results
    </h1>
    <div
      class="jsx-3550297800 movieContainer"
    >
      <a
        class="jsx-3550297800 movieItem"
        href="/movies/1"
      >
        <img
          class="jsx-3550297800"
          height="513px"
          src="https://image.tmdb.org/t/p/w342/1234"
          width="342px"
        />
        <p
          class="jsx-3550297800 movieTitle"
        >
          Back to the future
        </p>
      </a>
      <a
        class="jsx-3550297800 movieItem"
        href="/movies/2"
      >
        <img
          class="jsx-3550297800"
          height="513px"
          src="https://image.tmdb.org/t/p/w342/1234"
          width="342px"
        />
        <p
          class="jsx-3550297800 movieTitle"
        >
          Star Wars
        </p>
      </a>
    </div>
    <button
      class="goback"
    >
      Go back
    </button>
  </main>
  <footer
    class="jsx-627519052 footer"
  >
    Copyright 2020 - Sebastian Golbert
  </footer>
</DocumentFragment>
`;
